plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/preferences/SSLCertificatesPreference.java

```diff
@@ -105,13 +106,19 @@ public void setAllowedByCertificate(X509Certificate certificate, boolean result)
 	private boolean isValid(String validity) {
 		int expiresOnIndex = validity.indexOf(X509CertificateParser.EXPIRES_ON_PREFIX);
 		if(expiresOnIndex >= 0) {
-			SimpleDateFormat dateFormat = new SimpleDateFormat(X509CertificateParser.DATE_FORMAT);
+			SimpleDateFormat dateFormat = new SimpleDateFormat(X509CertificateParser.DATE_FORMAT,  Locale.ENGLISH);
```

Comment:
to me the parsing should go into a single class, the X509CertificateParser, not spread across several ones. Imho this method should be moved into X509CertificateParser. SSLCertificatesPreference being in ui while the parser is in core hints to this too imho. In terms of separation of concerns we have here a class that handles preferences matters, but also parses certificate validity. You also see this in the fact that this PR has to change things in 2 locations, because parsing the certificate is spread across 2 classes